src/core/Akka/Dispatch/Dispatcher.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Threading;
+using Akka.Event;
+using Akka.Util;
+
+namespace Akka.Dispatch
+{
+    /// <summary>
+    /// The event-based <see cref="Dispatcher"/> binds a set of actors to a thread pool backed up
+    /// by a thread-safe queue.
+    /// 
+    /// The preferred way of creating dispatchers is to define them in configuration and use the 
+    /// <see cref="Dispatchers.Lookup"/> method.
+    /// </summary>
+    public class Dispatcher : MessageDispatcher
```

Comment:
Generic `Dispatcher` - runs anything that isn't a `PinnedDispatcher`. The following dispatcher types have all been consolidated into `ExecutorService` implementations which run inside this class:

* ThreadPoolDispatcher
* ForkJoinDispatcher
* TaskDispatcher
* CurrentSynchronizationContextDispatcher